### PR TITLE
ingest: raise visibility of ingest plugin documentation

### PR DESCRIPTION
**Pull request by jakelandis**
part of #33188

* adds a processor plugin link from the main reference 
* moves set security user to main documentation 
* updates x-pack security doc to point to main documenation

-----------
Updated processor page
image

Updated x-pack/stack doc
image

EDIT: updated description to match latest PR
EDIT2: updated description to match latest PR

### CODE REVIEW

**Comment by jakelandis** [comment]
requesting @eskibars review based no https://github.com/elastic/elasticsearch/issues/33188#issuecomment-416791886

**Comment by rjernst** [comment]
This seems like the wrong place to mention the set user processor, as it is not installed as a normal plugin; it comes with the default distribution. The others must be installed as normal elasticsearch plugins.

**Comment by jakelandis** [comment]
> This seems like the wrong place to mention the set user processor

good point ! Let me try something different... adding WIP label.

**Comment by jakelandis** [comment]
Latest attempt to raise visibility of the Set Security User Processor is to move the documentation for it to primary documentation. 

This works because it is now part of the default distribution and will just work if security is setup. I included an IMPORTANT section to help ensure that it is obvious that an authenticated user is required, which by inference means this only works with the default distribution. 

@lcawl @eskibars  - could I get your opinion on adding this documentation to the main documentation that is dependent on security enabled + default distribution ?

File: x-pack/docs/en/security/authorization/set-security-user.asciidoc [hunk 1]
```diff
@@ -9,53 +9,18 @@ To guarantee that a user reads only their own documents, it makes sense to set u
...
-`set_security_user` ingest processor can help.
+{ref}/ingest-node-set-security-user-processor.html[Set Security User Processor] ingest processor can help.
 
 NOTE: Document level security doesn't apply to write APIs. You must use unique
 ids for each user that uses the same index, otherwise they might overwrite other
 users' documents. The ingest processor just adds properties for the current
 authenticated user to the documents that are being indexed.
 
-The `set_security_user` processor attaches user-related details (such as
+{ref}/ingest-node-set-security-user-processor.html[Set Security User Processor] processor attaches user-related details (such as
```
Comment (lcawl) [inline x-pack/docs/en/security/authorization/set-security-user.asciidoc]:
```suggestion
The {ref}/ingest-node-set-security-user-processor.html[set security user processor] attaches user-related details (such as
```

**Review by lcawl — APPROVED** [review]
Moving this content makes sense to me. I submitted two minor suggestions, but think it would be good for @debadair to weigh in on the Elasticsearch table of contents.

**Comment by debadair** [comment]
@jakelandis Agree that moving the Set security user processor into the list of ingest processors makes sense. 

I'm less sure about adding the "Plugin Processors" topic after the Processors topic. I would be inclined to add that information to the top level Processors topic. I think the point you're trying to make is that there are the built-in processors described under Processes and additional ingest processors that can be installed as ES plugins? If so, I'd be inclined to add the folliowing section to the bottom of the Processors page:

```
[float]
=== Ingest Processor Plugins

Additional ingest processors can be implemented and installed as Elasticsearch {plugins}/intro.html[plugins]. See {plugins}/ingest.html[Ingest plugins] for information about the available ingest plugins.
```

That way, people aren't confused about the difference between "Processors" and "Plugin Processors" and (hopefully) will avoid an extra click.

**Comment by jakelandis** [comment]
@lcawl @debadair thanks for the reviews ! changes as requested made, and screen shots of the rendered output are updated in the description.

**Comment by lcawl** [comment]
@jakelandis I ran the documentation builds and the changes look good to me.

**Review by eskibars — APPROVED** [review]

**Comment by eskibars** [comment]
LGTM!

**Pull request merged**